lib/module.js

```diff
@@ -435,7 +435,7 @@ Module._load = function(request, parent, isMain) {
     if (experimentalModules) {
       if (filename === null || /\.mjs$/.test(filename)) {
```

Comment:
Is there a reason why regex is used here instead of `filename.endsWith('.mjs')`?

Reply:
@bmack?

Reply:
@bmeck?

Reply:
Habits from before that existed. Technically we could/should not be using prototype functions since they could be tainted by preloading via `--require`.

Reply:
Ack.

Reply:
>  Technically we could/should not be using prototype functions since they could be tainted by preloading

There's a lot to tweak then because there's heavy use of `charAt`, `concat`, `pop`, `push`, `test`, `slice`, and on and on. It'd be neat to see if some of this could be solved with a new context to avoid plucking all these methods off.

Reply:
> There's a lot to tweak then because there's heavy use of charAt, concat, pop, push, test, slice, and on and on.

This concern pops up now and then.
Refs: https://github.com/nodejs/node/issues/12956
Refs: https://github.com/nodejs/node/pull/12981

Reply:
@jdalton Contexts are expensive w/o snapshots to create and we would need a frozen one to avoid mutations which would mean making that custom Snapshot, we couldn't use `vm.createContext` due to the funky sandbox bridge it creates leaking : https://github.com/nodejs/node/issues/6283 . I'd be open to using one if we had a snapshot, but it would still be ~10ms on my benchmarks to create a full one.